tee-worker/omni-executor/parentchain/listener/src/rpc_client.rs

```diff
@@ -72,11 +81,36 @@ impl<ChainConfig: Config> SubstrateRpcClient for SubxtClient<ChainConfig> {
 		}
 	}
 
-	async fn get_raw_metadata(&mut self, block_num: u64) -> Result<Vec<u8>, ()> {
-		let maybe_hash =
-			self.legacy.chain_get_block_hash(Some(block_num.into())).await.map_err(|_| ())?;
+	async fn get_raw_metadata(&mut self, block_num: Option<u64>) -> Result<Vec<u8>, ()> {
+		let maybe_hash = self
+			.legacy
```

Comment:
Out of curiosity, what are non-legacy methods for such queries? 😲

Reply:
I think it's `Metadata_metadata_at_version`